From 7f1034b4aebbc157581492daef1c19096e2c6abf Mon Sep 17 00:00:00 2001
From: stone.liu <stone.liu@dniserver.dnish.net>
Date: Wed, 26 Oct 2011 14:29:19 -0400
Subject: [PATCH 078/137] Define new variables for supporting SNATP2P in conntrack module

Define new variables for supporting SNATP2P in conntrack module.

Signed-off-by: Haiyue.Wang <haiyue@dniserver.dnish.net>
Signed-off-by: Jonathan Chang <jonathan@dniserver.dnish.net>

Conflicts:

	include/net/netfilter/nf_nat.h
	include/net/netns/ipv4.h

Signed-off-by: stone.liu <stone.liu@dniserver.dnish.net>
---
 include/net/netfilter/nf_nat.h |    4 ++++
 include/net/netns/ipv4.h       |    2 ++
 2 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/include/net/netfilter/nf_nat.h b/include/net/netfilter/nf_nat.h
index a9f72b1..eb0b8b0 100644
--- a/include/net/netfilter/nf_nat.h
+++ b/include/net/netfilter/nf_nat.h
@@ -69,7 +69,9 @@ struct nf_conn;
 struct nf_conn_nat
 {
 	struct hlist_node bysource;
+#if defined(CONFIG_IP_NF_TARGET_SNATP2P) || defined(CONFIG_IP_NF_TARGET_SNATP2P_MODULE)
 	struct hlist_node by_modified_source;
+#endif
 	struct nf_nat_seq seq[IP_CT_DIR_MAX];
 	struct nf_conn *ct;
 	union nf_conntrack_nat_help help;
@@ -93,9 +95,11 @@ extern unsigned int nf_nat_setup_info(struct nf_conn *ct,
 extern int nf_nat_used_tuple(const struct nf_conntrack_tuple *tuple,
 			     const struct nf_conn *ignored_conntrack);
 
+#if defined(CONFIG_IP_NF_TARGET_SNATP2P) || defined(CONFIG_IP_NF_TARGET_SNATP2P_MODULE)
 extern int find_appropriate_p2p_dst(struct net *net,
 				    const struct nf_conntrack_tuple *tuple,
 				    struct nf_conntrack_tuple *result);
+#endif
 
 static inline struct nf_conn_nat *nfct_nat(const struct nf_conn *ct)
 {
diff --git a/include/net/netns/ipv4.h b/include/net/netns/ipv4.h
index 76557fc..f4c649f 100644
--- a/include/net/netns/ipv4.h
+++ b/include/net/netns/ipv4.h
@@ -40,8 +40,10 @@ struct netns_ipv4 {
 	struct xt_table		*iptable_security;
 	struct xt_table		*nat_table;
 	struct hlist_head	*nat_bysource;
+	#if defined(CONFIG_IP_NF_TARGET_SNATP2P) || defined(CONFIG_IP_NF_TARGET_SNATP2P_MODULE)
 	struct hlist_head	*nat_by_modified_source;
 	int			nat_modified_vmalloced;
+	#endif
 	int			nat_vmalloced;
 #endif
 
-- 
1.7.4.1

